test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs

```diff
@@ -405,12 +415,28 @@ private static void DeleteDirectory(string path)
             }
             catch (UnauthorizedAccessException)
             {
-                Directory.Delete(path, true);
+                // Ignore UnauthorizedAccessException for the first attempts
+                var MaxTries = 100;
+
+                for (var i = 0; i < MaxTries; i++)
+                {
+                    
+                    try
+                    {
+                        Directory.Delete(path, true);
+
+                        break;
+                    }
+                    catch (Exception ex) when ((i < (MaxTries - 1)) && (ex is UnauthorizedAccessException))
```

Comment:
Why not catch the exception type you want rather than cast it?

Reply:
It was copied from Heng's branch so I didn't want merge conflicts, but I can fix it up :) 
Few people provided similar feedback on the method, so worth fixing :)